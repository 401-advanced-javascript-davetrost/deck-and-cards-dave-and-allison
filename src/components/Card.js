import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ title, image, content }) => (
  <div>
    <header><h3>{title}</h3></header>
    <figure>
      <img src={image} alt={title} />
    </figure>
    {content && <p>{content}</p>}
  </div>
);

Card.propTypes = {
  title: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  content: PropTypes.string
};

export default Card;
