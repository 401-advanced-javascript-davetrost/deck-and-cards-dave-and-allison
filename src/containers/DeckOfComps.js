import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class DeckOfComps extends Component {

  render() {
    const elements = this.props.items.map((item, i) => (
      <this.props.Component
        key={item._id || i}
        {...item} />
    ));

    return (
      <section>
        <header>
          <h2>{this.props.title}</h2>
        </header>
        {elements}
      </section>
    );
  }

  static propTypes = {
    title: PropTypes.string,
    items: PropTypes.arrayOf(PropTypes.object).isRequired,
    Component: PropTypes.oneOfType([PropTypes.func, PropTypes.object]).isRequired
  };
}
