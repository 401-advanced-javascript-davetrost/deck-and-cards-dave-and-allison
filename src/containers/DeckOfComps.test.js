import React from 'react';
import { shallow } from 'enzyme';
import DeckOfComps from './DeckOfComps';
import Card from '../components/Card';

describe('DeckOfComps component', () => {
  const cardExample = {
    title:'Test Title',
    image:'test image',
    content:'this is a test'
  };
  it('renders', () => {
    const wrapper = shallow(
      <DeckOfComps title="Test Title" items={[cardExample, cardExample]} Component={Card}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
