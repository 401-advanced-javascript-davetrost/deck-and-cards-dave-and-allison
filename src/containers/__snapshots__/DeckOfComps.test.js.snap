// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeckOfComps component renders 1`] = `
<section>
  <header>
    <h2>
      Test Title
    </h2>
  </header>
  <Card
    content="this is a test"
    image="test image"
    key="0"
    title="Test Title"
  />
  <Card
    content="this is a test"
    image="test image"
    key="1"
    title="Test Title"
  />
</section>
`;
